### PR TITLE
fix: [PIE-4556]: proper inferred types instead of any on column 

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary

proper inferred types instead of any on column 

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx [hunk 1]
```diff
@@ -48,10 +48,17 @@ import executionFactory from '@pipeline/factories/ExecutionFactory'
...
 import css from './ExecutionListTable.module.scss'
 
-type CellType = Renderer<CellProps<PipelineExecutionSummary>>
-type ColumnWithActions = { isPipelineInvalid: boolean; onViewCompiledYaml: () => void }
+type CellTypeWithActions<D extends Record<string, any>, V = any> = TableInstance<D> & {
+  column: ColumnInstance<D> & ExecutionListColumnActions
+  row: Row<D>
+  cell: Cell<D, V>
+  value: CellValue<V>
+}
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx:59]: I don't see a better way of inferring, until migration to react table 3 that has inbuilt and proper types for everything with table

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 4 Issues 
* Code issue 4 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10614)

**Comment by udayvunnam** [comment]
retrigger titlecheck

**Comment by udayvunnam** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx (hunk 1 above)
Comment (udayvunnam) [reply src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx:60]: @dhruv-harness addressed the comment from some old PR to have better types

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by dhruvUpadhyay07 — APPROVED** [review]

**Comment by udayvunnam** [comment]
@kapilkumawat86 can you please FM this? I will add a PR for coverage with different PR once new executions changes are done

**Pull request merged**